frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NodeVersionParser.java

```diff
@@ -0,0 +1,187 @@
+package com.github.eirslett.maven.plugins.frontend.lib;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.Arrays.asList;
+
+public class NodeVersionParser {
+
+    @VisibleForTesting
+    static final Set<String> UNUSUAL_VALID_VERSIONS;
+
+    @VisibleForTesting
+    static final Pattern VALID_VERSION_PATTERN = Pattern.compile("^v?\\d*\\.\\d*\\.\\d*$");
+
+    static {
+        UNUSUAL_VALID_VERSIONS = new HashSet<>();
+        UNUSUAL_VALID_VERSIONS.addAll(asList(
+                "latest",
```

Comment:
NIT(feature request): populate that list automatically based on some good external source

Reply:
This list only exists to allow aliases and weird historical versions working. This is our fork and neither of them are needed for our use-case.

Having a static list keeps this working offline, but tradeoff working with newer aliases.

This is the simplest solution to enable version validation which improves the lives of users more than aliases IMO.

If external parties find this fork useful and start using it, we're happy for them to design and build a better solution.

Reply:
I wonder if we need this list at all? (ie. do we actually have projects using these names, the names are of supported versions we would want someone to use atp etc?) (maybe "latest-vX" would be useful but for it a more generic parsing method might be viable?)